src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1385,111 +1446,209 @@ impl Cage {
                     match &*inodeobj {
+                        // For `File` type inode, the reading happens from the current position
+                        // of the object pointed by `position`. The fileobject is fetched from
+                        // the FileObjectTable and we start reading into the buffer `buf` until
+                        // `count` number of bytes.
                         Inode::File(_) => {
+                            // Get the current position of the File Descriptor Object.
                             let position = normalfile_filedesc_obj.position;
+                            // Get the file object associated with the file descriptor object
                             let fileobject =
                                 FILEOBJECTTABLE.get(&normalfile_filedesc_obj.inode).unwrap();
 
-                            if let Ok(bytesread) = fileobject.readat(buf, count, position) {
-                                //move position forward by the number of bytes we've read
-
-                                normalfile_filedesc_obj.position += bytesread;
-                                bytesread as i32
-                            } else {
-                                0 //0 bytes read, but not an error value that
-                                  // can/should be passed to the user
-                            }
+                            // `readat` function reads from file at specified offset into provided
+                            // C-buffer. If successful, then the position in the file descriptor
+                            // object is updated by adding the number of bytes read (bytesread).
+                            // This ensures that the next read operation will start from the correct
+                            // position.
+                            let bytesread = fileobject.readat(buf, count, position as usize).unwrap();
+                            //move position forward by the number of bytes we've read
+                            normalfile_filedesc_obj.position += bytesread;
+                            // Return the number of bytes read.
+                            bytesread as i32
                         }
 
+                        // For `CharDev` type inode, the reading happens from the Character Device
+                        // file, with each device type returning different results returned
+                        // from the `_read_chr_file` file.
                         Inode::CharDev(char_inode_obj) => {
+                            // reads from character devices by matching the device number (DevNo) of
+                            // the DeviceInode.
                             self._read_chr_file(&char_inode_obj, buf, count)
                         }
 
+                        // For `Socket` type inode, a panic is returned as socket type files are not
+                        // supported.
                         Inode::Socket(_) => {
```

Comment:
This really isn't "unsupported". It's actually failing a sanity check. a File type fd should never have a socket type inode

Reply:
updated